generators/server/templates/src/test/java/package/web/rest/KafkaResourceIT.java.ejs

```diff
@@ -18,156 +18,85 @@
 -%>
 package <%= packageName %>.web.rest;
 
-import <%= packageName %>.config.KafkaProperties;
-import org.apache.kafka.clients.consumer.ConsumerRecord;
-import org.apache.kafka.clients.consumer.ConsumerRecords;
-import org.apache.kafka.clients.consumer.KafkaConsumer;
-import org.apache.kafka.clients.producer.KafkaProducer;
-import org.apache.kafka.clients.producer.ProducerRecord;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
-import org.springframework.http.MediaType;
-<%_ if (reactive) { _%>
-import org.springframework.test.web.reactive.server.WebTestClient;
-<%_ } else { _%>
-import org.springframework.test.web.servlet.MockMvc;
-import org.springframework.test.web.servlet.MvcResult;
-import org.springframework.test.web.servlet.setup.MockMvcBuilders;
-<%_ } _%>
-import org.testcontainers.containers.KafkaContainer;
-import org.testcontainers.utility.DockerImageName;
-
-import java.time.Duration;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
-
 import static org.assertj.core.api.Assertions.assertThat;
-<%_ if (!reactive) { _%>
 import static org.junit.jupiter.api.Assertions.fail;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
-<%_ } _%>
-
-class <%= upperFirstCamelCase(baseName) %>KafkaResourceIT {
 
-    private static boolean started = false;
-    private static KafkaContainer kafkaContainer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
 
-<%_ if (reactive) { _%>
-    private WebTestClient client;
-<%_ } else { _%>
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.util.MimeTypeUtils;
+import <%= packageName %>.IntegrationTest;
+import <%= packageName %>.config.EmbeddedKafka;
+import <%= packageName %>.config.KafkaSseConsumer;
+import <%= packageName %>.config.KafkaSseProducer;
+
+@IntegrationTest
+@AutoConfigureMockMvc
+@WithMockUser
+@EmbeddedKafka
+class <%= capitalizedBasename %>KafkaResourceIT {
```

Comment:
```suggestion
class <%= capitalizedBaseName %>KafkaResourceIT {
```

Reply:
<%= upperFirstCamelCase(baseName) %> ^^. I'm on it